crates/worker/src/partition/state_machine/lifecycle/migrate_journal_table.rs

```diff
@@ -67,10 +68,13 @@ where
                 journal_table_v1::JournalTable::delete_journal(ctx.storage, &self.invocation_id, 1)
                     .await
                     .map_err(Error::Storage)?;
+
+                // Mutate the journal metadata commands
+                self.metadata.journal_metadata.commands = 1;
             } else {
                 // We're already in journal table v2, nothing to migrate!!!
             }
-        } else if self.journal_length >= 1 {
+        } else if self.metadata.journal_metadata.length >= 1 {
             // We can just check corruption here.
             // Length can be greater than 1 when we have either Completions (in the old table) or Notifications (in the new table).
             // Because of the different Awakeable id format, we cannot incur in the situation where we write to the old table for a Completion arrived before the pinned deployment.
```

Comment:
Can the `debug_assert` become an `assert` in the line below? I think we should have these kind of checks also in release builds.